### ISSUE TITLE
Code discussion - PostgreSQL / MySQL password config

### ISSUE THREAD

**Issue by consideRatio** [jupyterhub/zero-to-jupyterhub-k8s#1100]
I was and still am a bit uncertain if the following is something we can improve the code on.
- We should avoid asking for duplicated inputs in the configuration, but in this case it may have avoided a lot of trouble, but I don't know. Does it make sense to ask the user to repeat the `hub.db.password` in the `hub.db.url` configuration?
- Does JupyterHub happen to look for the MYSQL_PWD and PGPASSWORD in another place than in the environment variables? If so, perhaps we should attempt to make it available like that instead. Does this improve security? Hmmm... Eh... Confused about this... Why did I think it would do that? Perhaps that it will be hidden from a `config describe pod hub` command? Or is this listed in a `describe` ? Hmm...

In the z2jh [configuration reference](https://zero-to-jupyterhub.readthedocs.io/en/latest/reference.html#hub-db), I read about `hub.db.password`, but also that `hub.db.url` that contained that password in the connection string.

So I inspected how they are utilized.
- `hub.db.password` is used to store itself in the `hub-secret`, and an ENV variable is created on the hub pod through with this secret as a reference.

```yaml
            {{- if eq .Values.hub.db.type "mysql" }}
            - name: MYSQL_PWD
              valueFrom:
                secretKeyRef:
                  name: hub-secret
                  key: hub.db.password
            {{- else if eq .Values.hub.db.type "postgres" }}
            - name: PGPASSWORD
              valueFrom:
                secretKeyRef:
                  name: hub-secret
                  key: hub.db.password
            {{- end }}
```

- `hub.db.url` is used to configure `c.JupyterHub.db_url` from the `jupyterhub_config.py` in the hub image.
```python
# configure the hub db connection
db_type = get_config('hub.db.type')
if db_type == 'sqlite-pvc':
    c.JupyterHub.db_url = "sqlite:///jupyterhub.sqlite"
elif db_type == "sqlite-memory":
    c.JupyterHub.db_url = "sqlite://"
else:
    set_config_if_not_none(c.JupyterHub, "db_url", "hub.db.url")
```

**Comment by minrk** [comment]
>  Does it make sense to ask the user to repeat the hub.db.password in the hub.db.url configuration?

No, it shouldn't need to be in the hub.db.url. It should only be in hub.db.password. Though putting it in db.url works fine.

> Does JupyterHub happen to look for the MYSQL_PWD and PGPASSWORD in another place than in the environment variables?

It is read from the env, but by the underlying db client library, not jupyterhub.

> Does this improve security?

Somewhat, in that it goes to the hub env via a secret instead of configmap, etc. 

> Perhaps that it will be hidden from a config describe pod hub command?

Yes, describe shows:

```yaml
Environment:
      PYTHONUNBUFFERED:        1
      HELM_RELEASE_NAME:       hub
      JPY_COOKIE_SECRET:       <set to the key 'hub.cookie-secret' in secret 'hub-secret'>  Optional: false
      POD_NAMESPACE:           default (v1:metadata.namespace)
      CONFIGPROXY_AUTH_TOKEN:  <set to the key 'proxy.token' in secret 'hub-secret'>      Optional: false
      PGPASSWORD:              <set to the key 'hub.db.password' in secret 'hub-secret'>  Optional: false
```

To test, I ran:

```
helm install stable/postgresql --name pgdb -f pg.yaml
```

with pg.yaml:

```yaml
postgresqlPassword: "secretsss"
postgresqlDatabase: "jupyterhub"
postgresqlUsername: "jupyterhubpg"
```

and jupyterhub config:

```yaml
hub:
  db:
    type: postgres
    password: "secretsss"
    url: "postgresql+psycopg2://jupyterhubpg@pgdb-postgresql/jupyterhub"
```

it successfully authenticated.

To summarize, the only thing that should change is updating the docs for db.url to avoid the implication that the password should go in the url (even though it will still work this way, it is slightly less secure).
